test/attr/attr_escaping.swift

```diff
@@ -160,3 +160,20 @@ class FooClass {
     set(newValue) { stored = newValue } // expected-error{{cannot assign value of type '(@escaping () -> Int) -> Void' to type 'Optional<(() -> Int) -> Void>'}}
   }
 }
+
+// A call of a closure literal should be non-escaping
+func takesInOut(y: inout Int) {
+  {
+    y += 1 // no-error
+  }()
+}
+
+class HasIVarCaptures {
+  var x: Int = 0
+
+  func method() {
+    {
+      x += 1 // no-error
+    }()
```

Comment:
You're right, those test cases were bogus and always passed anyway. I added more detailed tests and fixed the other bug we talked about with capture lists.